qiskit/ignis/experiments/calibration/instruction_data/compiler.py

```diff
@@ -0,0 +1,318 @@
+            return channel(index=index)
+
+        raise Exception('Channel name {name} is not correct syntax.'.format(name=ch_str))
+
+    def visit(self, node):
+        """Visit a node."""
+        method = 'visit_' + node.__class__.__name__
+        visitor = getattr(self, method)
+
+        return visitor(node)
+
+    def visit_PulseInst(self, node: PulseInst) -> pulse.Instruction:
+        """Evaluate pulse instruction node and return play instruction.
+
+        Args:
+            node: PulseInst node to evaluate.
+
+        Returns:
+            Play instruction.
+        """
+        parametric_pulse = self._inst_def.parametric_shapes[node.shape].value
+
+        # generate parameter names on the fly
+        parametric_pulse_kwargs = dict()
+        for pname in utils.get_pulse_parameters(parametric_pulse):
+            scoped_pname = utils.add_scope(pname, node.channel, node.name)
+            # TODO calling parameter biding here is bit strange.
+            # This parameter binding should be offloaded to the get_gate_schedule method.
+            # However, we cannot do this now because of parametrization of duration.
+            # The role of parser is to create program, not the parameter-bound schedule.
+            param_val = self._inst_def.pulse_parameter_table.get_parameter(
+                parameter=scoped_pname,
+                gate_id=self._gate_id,
+                series=self._inst_def.series
+            )
+            if param_val is not None and scoped_pname not in self._free_parameters:
+                parametric_pulse_kwargs[pname] = param_val
+            else:
+                parametric_pulse_kwargs[pname] = circuit.Parameter(scoped_pname)
+
+        # pulse name for visualization purpose
+        pulse_name = '{}.{}.{}'.format(node.name, node.channel, self._gate_id)
+        played_pulse = parametric_pulse(**parametric_pulse_kwargs, name=pulse_name)
+        channel = self._get_channel(node.channel)
+
+        return pulse.Play(played_pulse, channel)
+
+    def visit_FrameInst(self, node: FrameInst) -> pulse.Instruction:
+        """Evaluate frame instruction node and return frame change instruction.
+
+        Args:
+            node: FrameInst node to evaluate.
+
+        Returns:
+            Arbitrary frame change instruction specified by AST name.
+        """
+        raise NotImplementedError
+
+    def visit_Reference(self, node: Reference) -> pulse.Schedule:
+        """Evaluate reference node and return arbitrary schedule.
+
+        Args:
+            node: Reference node to evaluate.
+
+        Returns:
+            Arbitrary schedule from another schedule database entry.
+        """
+        return self._inst_def.get_gate_schedule(
+            gate_id=node.name,
+            free_parameter_names=self._free_parameters
+        )
+
+    def visit_ScheduleBlock(self, node: ScheduleBlock) -> pulse.Schedule:
+        """Evaluate schedule block node and return arbitrary schedule.
```

Comment:
Why are we returning an **arbitrary** schedule? What does that mean?

Reply:
This is bad wording... _schedule defined by input code_?

Reply:
```suggestion
        """Return the pulse Schedule defined by the input ScheduleBlock.
```